master/buildbot/data/buildrequests.py

```diff
@@ -29,6 +30,28 @@
 
 class Db2DataMixin(object):
 
+    def _generate_filtered_properties(self, props, filters):
+        """
+        This method returns Build's properties according to property filters.
+
+        :param props: Properties as a dict (from db)
+        :param filters: Desired properties keys as a list (from API URI)
+        """
+        # by default no properties are returned
+        if props and filters:  # pragma: no cover
+            return (props
+                    if '*' in filters
+                    else dict(((k, v) for k, v in iteritems(props) if k in filters)))
+
+    @defer.inlineCallbacks
+    def addPropertiesToBuildRequest(self, buildrequest, filters):
+        if not filters:
+            return
+        props = yield self.master.db.buildsets.getBuildsetProperties(buildrequest['buildsetid'])
```

Comment:
Those lines are not covered. We need a unit test to test that buildsetproperties are correctly fetched.

Reply:
Added tests, please review again.